src/OrleansRuntime/Streams/QueueBalancer/DeploymentBasedQueueBalancer.cs

```diff
@@ -156,22 +161,22 @@ public bool SubscribeToQueueDistributionChangeEvents(IStreamQueueBalanceListener
             {
                 if (queueBalanceListeners.Contains(observer))
                 {
-                    return false;
+                    return Task.FromResult(false);
                 }
                 queueBalanceListeners.Add(observer);
-                return true;
+                return Task.FromResult(true);
             }
         }
 
-        public bool UnSubscribeToQueueDistributionChangeEvents(IStreamQueueBalanceListener observer)
+        public Task<bool> UnSubscribeToQueueDistributionChangeEvents(IStreamQueueBalanceListener observer)
```

Comment:
We should rename this: `UnsubscribeFromQueueBalancingEvents` and similar for the subscribe method